doc/source/dask-on-ray.rst

```diff
@@ -25,50 +25,73 @@ Here's an example:
 
    import ray
    from ray.util.dask import ray_dask_get
-   import dask.delayed
+   import dask.array as da
+   import dask.dataframe as dd
+   import numpy as np
+   import pandas as pd
    import time
 
    # Start Ray.
    # Tip: If you're connecting to an existing cluster, use ray.init(address="auto").
    ray.init()
 
+   d_arr = da.from_array(np.random.randint(0, 1000, size=(256, 256)))
 
-   @dask.delayed
-   def inc(x):
-       time.sleep(1)
-       return x + 1
+   # The Dask scheduler submits the underlying task graph to Ray.
+   d_arr.mean().compute(scheduler=ray_dask_get)
 
-   @dask.delayed
-   def add(x, y):
-       time.sleep(3)
-       return x + y
+   # Set the scheduler to ray_dask_get in your config so you don't have to specify it on
+   # each compute call.
+   dask.config.set(scheduler=ray_dask_get)
+
+   df = dd.from_pandas(pd.DataFrame(np.random.randint(0, 100, size=(1024, 2)), columns=["age", "grade"]))
+   df.groupby(["age"]).mean().compute()
 
-   x = inc(1)
-   y = inc(2)
-   z = add(x, y)
-   # The Dask scheduler submits the underlying task graph to Ray.
-   z.compute(scheduler=ray_dask_get)
 
 Why use Dask on Ray?
 
-   1. If you'd like to create data analyses using the familiar NumPy and Pandas
-      APIs provided by Dask and execute them on a production-ready distributed
-      task execution system like Ray.
-   2. If you'd like to use Dask and Ray libraries in the same application
-      without having two different task execution backends.
+   1. If you'd like to create data analyses using the familiar NumPy and Pandas APIs provided by Dask and execute them on a fast, fault-tolerant distributed task execution system geared towards production, like Ray.
+   2. If you'd like to use Dask and Ray libraries in the same application without having two different clusters.
    3. To take advantage of Ray-specific features such as the
```

Comment:
Please also mention the mature actor implementation. Also, let's put this as the first reason (since I believe it is the strongest benefit).

Reply:
That's a good motivation for using Ray in general, but Dask-on-Ray doesn't use actors, so I think that calling out such an orthogonal feature doesn't really fit here. We can point them to a more general "why use Ray" docs page that contains a more comprehensive listing of Ray's features, but I think that we should be calling out Ray features that Dask workloads will benefit from here.

Reply:
Hmm yeah a good point about actors. But shared memory is definitely a useful feature. Should we only mention shared memory in this case?

Reply:
Shared memory is mentioned, or are you saying that you think that shared memory should go first, before mentioning the cluster launcher?

Reply:
Yeah I think the first reason is the least compelling reason compared to other 2. Maybe we can change the order? 2 -> 3 -> 1 or 3 -> 2 -> 1